sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableChange.java

```diff
@@ -259,6 +298,69 @@ public int hashCode() {
     }
   }
 
+  interface ColumnPosition {
+
+    static ColumnPosition first() {
+      return First.SINGLETON;
+    }
+
+    static ColumnPosition after(String column) {
+      return new After(column);
+    }
+  }
+
+  /**
+   * Column position FIRST means the specified column should be the first column.
+   * Note that, the specified column may be a nested field, and then FIRST means this field should
+   * be the first one within the struct.
+   */
+  final class First implements ColumnPosition {
+    private static final First SINGLETON = new First();
```

Comment:
Hi, All.
Sorry for late nit-picking, but shall we rename this to `INSTANCE` for consistency with the other singletons?
During review a new PR, https://github.com/apache/spark/pull/27380, I found that we are starting to loss the consistency. I'll make a follow-up. 

cc @brkyvz